### PR TITLE
Facepile (v8): convert tests to use testing-library

### PR DESCRIPTION
**Pull request by TristanWatanabe**
## Current Behavior

- v8 `Facepile` tests don't use testing-library

## New Behavior

- Convert `Facepile` to fully use testing-library

## Related Issue(s)

part of #21663

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/af38b775a98bfbcd214ef489ffb1cd6b941d9444'>af38b775a98bfbcd214ef489ffb1cd6b941d9444</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/BaseButton-mount_base.html">990</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/BaseButton-mount.html">954</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Breadcrumb-mount_base.html">2811</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Breadcrumb-mount.html">2851</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Checkbox-mount_base.html">1551</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Checkbox-mount.html">1569</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/CheckboxBase-mount_base.html">1323</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/CheckboxBase-mount.html">1269</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ChoiceGroup-mount_base.html">4782</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ChoiceGroup-mount.html">4888</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ComboBox-mount_base.html">1009</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ComboBox-mount.html">1004</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/CommandBar-mount_base.html">10893</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/CommandBar-mount.html">10813</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ContextualMenu-mount_base.html">11952</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ContextualMenu-mount.html">11835</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/DefaultButton-mount_base.html">1154</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/DefaultButton-mount.html">1201</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/DetailsRow-mount_base.html">3908</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/DetailsRow-mount.html">4025</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/DetailsRowFast-mount_base.html">3962</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/DetailsRowFast-mount.html">3946</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/DetailsRowNoStyles-mount_base.html">3776</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/DetailsRowNoStyles-mount.html">3820</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Dialog-mount_base.html">2291</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Dialog-mount.html">2243</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/DocumentCardTitle-mount_base.html">174</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/DocumentCardTitle-mount.html">172</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Dropdown-mount_base.html">3384</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Dropdown-mount.html">3379</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/FocusTrapZone-mount_base.html">1894</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/FocusTrapZone-mount.html">1858</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/FocusZone-mount_base.html">1847</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/FocusZone-mount.html">1878</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/IconButton-mount_base.html">1847</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/IconButton-mount.html">1845</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Label-mount_base.html">361</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Label-mount.html">353</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Layer-mount_base.html">3042</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Layer-mount.html">3055</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Link-mount_base.html">462</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Link-mount.html">474</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/MenuButton-mount_base.html">1577</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/MenuButton-mount.html">1545</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/MessageBar-mount_base.html">2158</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/MessageBar-mount.html">2219</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Nav-mount_base.html">3368</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Nav-mount.html">3385</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/OverflowSet-mount_base.html">1103</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/OverflowSet-mount.html">1142</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Panel-mount_base.html">2232</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Panel-mount.html">2263</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Persona-mount_base.html">1053</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Persona-mount.html">1040</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Pivot-mount_base.html">1499</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Pivot-mount.html">1522</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/PrimaryButton-mount_base.html">1352</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/PrimaryButton-mount.html">1335</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Rating-mount_base.html">7942</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Rating-mount.html">7937</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/SearchBox-mount_base.html">1344</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/SearchBox-mount.html">1385</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Shimmer-mount_base.html">2508</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Shimmer-mount.html">2679</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Slider-mount_base.html">1944</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Slider-mount.html">1959</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/SpinButton-mount_base.html">5089</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/SpinButton-mount.html">5132</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Spinner-mount_base.html">461</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Spinner-mount.html">433</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/SplitButton-mount_base.html">3318</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/SplitButton-mount.html">3252</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Stack-mount_base.html">544</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Stack-mount.html">516</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/StackWithIntrinsicChildren-mount_base.html">2328</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/StackWithIntrinsicChildren-mount.html">2342</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/StackWithTextChildren-mount_base.html">5330</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/StackWithTextChildren-mount.html">5404</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/SwatchColorPicker-mount_base.html">11856</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/SwatchColorPicker-mount.html">11890</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/TagPicker-mount_base.html">2741</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/TagPicker-mount.html">2710</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/TeachingBubble-mount_base.html">92874</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/TeachingBubble-mount.html">92913</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Text-mount_base.html">426</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Text-mount.html">429</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/TextField-mount_base.html">1371</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/TextField-mount.html">1433</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ThemeProvider-mount_base.html">1174</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ThemeProvider-mount.html">1191</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ThemeProvider-virtual-rerender_base.html">666</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ThemeProvider-virtual-rerender.html">621</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1861</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1876</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Toggle-mount_base.html">781</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/Toggle-mount.html">786</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/buttonNative-mount_base.html">126</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/22282/perf-test/buttonNative-mount.html">121</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/Facepile/Facepile.test.tsx [hunk 1]
```diff
@@ -181,49 +176,46 @@ describe('Facepile', () => {
       />,
     );
 
-    expectOne(wrapper, '.ms-Facepile-addButton .ms-Persona-coin.ms-Persona--size32');
-    expectOne(wrapper, '.ms-Facepile-overflowButton .ms-Persona--size32');
+    const addButton = getAllByRole('button')[0];
+    const overflowButton = getAllByRole('button')[1];
```
Comment (ecraig12345) [inline packages/react/src/components/Facepile/Facepile.test.tsx:180]: nit
```suggestion
    const [addButton, overflowButton] = getAllByRole('button');
```

File: packages/react/src/components/Facepile/Facepile.test.tsx [hunk 2]
```diff
@@ -181,49 +176,46 @@ describe('Facepile', () => {
...
 
-    const faces = findNodes(wrapper, '.ms-Facepile-person .ms-Persona-coin');
+    expect(addButton.firstElementChild!.firstElementChild!.className).toContain('ms-Persona--size32');
```
Comment (ecraig12345) [inline packages/react/src/components/Facepile/Facepile.test.tsx:182]: Is there any way to get this that's less reliant on the current DOM structure, like by text or aria-label or even classname?

Comment (TristanWatanabe) [reply packages/react/src/components/Facepile/Facepile.test.tsx]: Unfortunately no text or aria-label, went with querying for a classname instead

File: packages/react/src/components/Facepile/Facepile.test.tsx [hunk 3]
```diff
@@ -181,49 +176,45 @@ describe('Facepile', () => {
       />,
     );
 
-    expectOne(wrapper, '.ms-Facepile-addButton .ms-Persona-coin.ms-Persona--size32');
-    expectOne(wrapper, '.ms-Facepile-overflowButton .ms-Persona--size32');
+    const [addButton, overflowButton] = getAllByRole('button');
 
-    const faces = findNodes(wrapper, '.ms-Facepile-person .ms-Persona-coin');
+    expect(addButton.querySelectorAll('.ms-Persona--size32')).toBeTruthy();
```
Comment (ecraig12345) [inline packages/react/src/components/Facepile/Facepile.test.tsx:181]: querySelectorAll always returns a NodeList, so you should probably test the length or something instead

**Review by ecraig12345 — APPROVED** [review]

**Pull request merged**